### PR TITLE
Chore/Unit Test for Legal Texts

### PR DESCRIPTION
**Pull request by tcfos**
This PR contains two unit tests to ensure that the legal texts for PL, RO, and BG are equal to EN.

There is a **dependency** to https://github.com/corona-warn-app/cwa-app-ios/pull/1398. Please merge #1398 before #1407

No Jira item. No screenshots.

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationLegalTextsTest.swift [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import Foundation
+import XCTest
+@testable import ENA
+
+class AppInformationLegalTextsTest: XCTestCase {
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationLegalTextsTest.swift:23]: I can't see the file in Xcode. Only in the file system.

File: src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationLegalTextsTest.swift [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+			XCTAssertEqual(targetText, sourceText)
+		}
+	}
+	
+	private func getText(fromFile: String, withExtension: String, localization: String) throws -> String {
+		let directory = localization + ".lproj"
+		guard let url = Bundle.main.url(forResource: fromFile, withExtension: withExtension) else {
+			return ""
+		}
+		let data = try Data(contentsOf: url.deletingLastPathComponent()
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationLegalTextsTest.swift:55]: Why is removing the path components and adding a directory needed?
Wouldn't it be enough to just Data(contentsOf: url) ?

File: src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationLegalTextsTest.swift (hunk 1 above)
Comment (tcfos) [reply src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationLegalTextsTest.swift:23]: Fixed. New file AppInformationLegalTextsTest.swift added to Xcode

File: src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationLegalTextsTest.swift (hunk 2 above)
Comment (tcfos) [reply src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationLegalTextsTest.swift:55]: The tests iterates over all target languages, and it has to read the localized files from different directories.

**Comment by tcfos** [comment]
SonarCloud check failed because of low coverage of AppInformationLegalTextsTest. But this is a unit test!

File: src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationLegalTextsTest.swift (hunk 2 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationLegalTextsTest.swift:55]: I would propose to use subdirectory api:

```
private func getText(fromFile: String, withExtension: String, localization: String) throws -> String {
 let directory = localization + ".lproj"
 guard let url = Bundle.main.url(forResource: fromFile, withExtension: withExtension, subdirectory: directory) else {
  return ""
 }
 let data = try Data(contentsOf: url)

 return String(data: data, encoding: .utf8) ?? ""
}
```

**Review by ArturFriesen — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationLegalTextsTest.swift (hunk 2 above)
Comment (tcfos) [reply src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationLegalTextsTest.swift]: Thanks for the feedback. Changed func getText() accordingly.

**Comment by tcfos** [comment]
**Caution** – please merge #1398 first because of dependencies.

**Review by marcussc — APPROVED** [review]

**Comment by tcfos** [comment]
PR cancelled due to dependencies. New PR will be created.

**Pull request closed without merging**